### PR TITLE
User profile

### LINKED ISSUE #199: #197 Implement a new profile view
Implement the new profile view, Don't add to real UI yet.

### PR DESCRIPTION
**Pull request by aolsson711**
- Created the new profile activity. 
- No functionality added, only buttons add hide correct buttons
- No tests added since functionality will create meaningful tests.

closes #199

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    /**
+     * Setup profile text
+     * @param username  of profile
+     * @param points of profile
+     */
+    private void setupProfile(String username, int points){
+        TextView usernameText = findViewById(R.id.profile_username);
+        TextView pointsText = findViewById(R.id.profile_points);
+
+        String profileText = "Score: " + UIUtils.IntegerConvert(points);
```
Comment (bgabriel998) [inline app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java:54]: Use string ressources for translations

File: app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        usernameText.setText(username);
+        pointsText.setText(profileText);
+    }
+
+    /**
+     * Hide the correct views based on if the profile is the user itself
+     * or if it is a friend or not of the user.
+     * @param self
+     * @param friends
```
Comment (bgabriel998) [inline app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java:63]: Parameters description ?

File: app/src/main/java/ch/epfl/sdp/peakar/utils/UIUtils.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+public class UIUtils {
+
+    private final static int INTEGER_CHUNK = 3;
+    private final static String SPACE = " ";
+
+    /**
+     * Convert an given integer into a string follow the pattern of
+     * 12 345 678
+     * @return String representing int.
+     */
```
Comment (bgabriel998) [inline app/src/main/java/ch/epfl/sdp/peakar/utils/UIUtils.java:12]: Parameter dscription is missing
```suggestion
     * @param number. Integer number to convert to string
     */
```

**Review by bgabriel998 — COMMENTED** [review]
Good one. Small modifications requested before merging.

File: app/src/main/java/ch/epfl/sdp/peakar/utils/UIUtils.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package ch.epfl.sdp.peakar.utils;
+
+public class UIUtils {
```
Comment (giommok) [inline app/src/main/java/ch/epfl/sdp/peakar/utils/UIUtils.java:3]: Even if the name is quite self-explanatory, should we add a brief description of this class?

**Review by giommok — COMMENTED** [review]
Good job!
I left a comment because I noticed you have created a `UIUtils` class and I thought that maybe we could add a short JavaDoc comment to explain what we should put in there.

**Review by giommok — APPROVED** [review]

**Review by bgabriel998 — CHANGES_REQUESTED** [review]
Following files decrease the coverage by 3%... 
image



--> EDIT  : resolved on discord

**Review by bgabriel998 — APPROVED** [review]

**Comment by bgabriel998** [comment]
The test coverage is decreased but will be improved as soon as the new UI will be implemented with real data. Yet only mock data is used

**Pull request merged**